### DISCUSSION TITLE
Querying occluded triangles

### DISCUSSION THREAD

**Discussion by nikhil-convrseai** [gpuweb/gpuweb#4870] [Q&A] [answered]
I'm trying to build a tool in which one can identify if there are any occluded triangles  in the whole 3D scene. Please note that I'm not asking about occluded objects as that I'm able to do.

In opengl there is an occlusion query which helps in telling how many samples passed the test exactly, is there something similar in WebGPU as well? 

If yes, how can we access it? 

If not, then how can I help in contributing for it?

**Comment by kainino0x** [comment]
Yes, WebGPU has occlusion queries. Here is the spec section, you can also just ctrl-F for `occlusion` on the page.
https://gpuweb.github.io/gpuweb/#occlusion

**Reply by kainino0x** [reply, marked as answer]
> In opengl there is an occlusion query which helps in telling how many samples passed the test exactly, is there something similar in WebGPU as well?

Sorry, I forgot to mention we _don't_ support "precise" occlusion queries. Please comment here if you need them:
https://github.com/gpuweb/gpuweb/issues/3715
